### PR TITLE
refactor(DataSpreadsheet): implement typescript types

### LINKED ISSUE #5118: Add Typescript types to Dataspreadsheet

### PR DESCRIPTION
**Pull request by makafsal**
Closes #5118 #5119 #5120 

#### What did you change?
`packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx`
`packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheetHeader.tsx`
`packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheetBody.tsx`

#### How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 1]
```diff
@@ -62,14 +65,102 @@ const defaults = {
...
+
 /**
  * DataSpreadsheet: used to organize and display large amounts of structured data, separated by columns and rows in a grid-like format.
  */
 export let DataSpreadsheet = React.forwardRef(
   (
     {
       // The component props, in alphabetical order (for consistency).
-      cellSize = defaults.cellSize,
+      cellSize = 'sm',
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:163]: i believe this should still be `defults.cellSize`

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks good. just one small comment!

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 1 above)
Comment (makafsal) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:162]: The type of `cellSize` changed to `type Size = 'xs' | 'sm' | 'md' | 'lg'`, hence we can define the default `cellSize` as one of type `Size` not a string type.

**Review by matthewgallo — CHANGES_REQUESTED** [review]
Great job on this! This is a huge component with a _lot_ of functionality. Can you break out the types for `ActiveCellCoordinates`, `Column` and possibly even `PrevState` into a separate type utility and use them across the `DataSpreadsheet`, `DataSpreadsheetBody`, and `DataSpreadsheetHeader` to reduce duplication?

**Review by matthewgallo — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**